### DISCUSSION TITLE
Offline Exchange with React Native

### DISCUSSION THREAD

**Discussion by BjoernRave** [urql-graphql/urql#1286] [Q&A] [answered]
I want to use the `offlineExchange` with react-native. 

when implementing the storage fn, I came upon a problem, where I am not sure if it matters:

In the docs it says:
```
writeMetadata : (json: SerializedRequest[]) => void
```

However since I am using `AsyncStorage` all operations are promises, so my function looks like this: `(json: SerializedRequest[]) => Promise<void>` . Is this a problem?

**Comment by kitten** [comment, marked as answer]
This is a fire-and-forget function, so as long as you ensure that that data is written it doesn't matter. Essentially the function signature is telling you here that you'll have to make sure that the metadata is written but Graphcache will never wait for any written data to be committed, since it never reads this data back outside of its startup routine.
